### ISSUE TITLE
[CI] PR failing to link mme-tests against libyaml-cpp-dev in Focal Dockerfile

### ISSUE THREAD

**Issue by electronjoe** [magma/magma#5891]
Yogesh Prandey [reported a CI failure for a PR](https://app.circleci.com/pipelines/github/magma/magma/18316/workflows/065e6b62-db65-4e30-b465-a8ee5c75a11d/jobs/177755).

The issue appears at face value to be a linking problem against `libyaml-cpp-dev`.

**Comment by electronjoe** [comment]
We have not pinned the version of libyaml-cpp-dev, I would like to know it is stable / has not changed.

My docker image built recently from Dockerfile.ubunto20.04 and successfully running tests.

```shell
root@c23723beb780:/# apt list libyaml-cpp-dev
Listing... Done
libyaml-cpp-dev/focal,now 0.6.2-4ubuntu1 amd64 [installed]
```

It's possible the PR contains a test / linking problem - and there is no issue with our Docker container. Still to be determined.

**Comment by electronjoe** [comment]
For context, our established Ubuntu 18.04 VM has the following.

```
vagrant@magma-dev:~$ apt list libyaml-cpp-dev
Listing... Done
libyaml-cpp-dev/oldstable,now 0.5.2-4 amd64 [installed]
```

Note that our new Focal based Vagrant VM has the same version as our Docker container (not surprising):

```shell
vagrant@magma-dev-focal:~$ apt list libyaml-cpp-dev
Listing... Done
libyaml-cpp-dev/focal,now 0.6.2-4ubuntu1 amd64 [installed]
```

So I am confident (as I just built the focal) that our libyaml-cpp-dev version has not moved on us.

So the issue here is restricted to something along the lines of.

- New dependencies with invalid build ordering or missing includes somewhere in our source

I cannot think of anything else at this point.

So either an issue with missing includes in magma. Or an issue with CMake configs (existing in Magma and now imported / used in a way that highlights failure, or issues with new CMake configs).

**Comment by electronjoe** [comment]
My present concern is that a missing header include (likely in Magma) is causing this problem - this is a frequent occurrence in the new build environment (new CMake version, etc).  @themarwhal suggested we try fixing up the includes for the orc8r/gateway/c targets as the linking errors seem to stem from service_registry build and linking in this PR.  I'm going to explore that now (and cherry-pick this PR to validate or invalidate fixes).  These missing header includes are rampant and make development brittle, particularly as we move forward our build toolchains. See #4868 and IWYU tool recently committed in #5761.

**Comment by electronjoe** [comment]
@panyogesh appears to have resolved this in the PR as follows, though I'm not certain this should have been necessary (if not points to a CMake broader issue?).

> the issue got resolved by adding -lyaml-cpp in the new test module added.. I have updated the commit.

**Comment by electronjoe** [comment]
Regarding missing header includes / extraneous includes in the context of this PR. I merged #5895 this morning to address these.

**Comment by electronjoe** [comment]
We should circle back on the Linker needs here..  But I think pragmatically it lacks immediate prioritization.
